src/uct/ib/base/ib_md.c

```diff
@@ -841,9 +873,38 @@ static ucs_status_t uct_ib_mem_alloc(uct_md_h uct_md, size_t *length_p,
     length     = *length_p;
     exp_access = uct_ib_md_access_flags(md, flags, length) |
                  IBV_EXP_ACCESS_ALLOCATE_MR;
-    status = uct_ib_md_reg_mr(md, NULL, length, exp_access, 0, &memh->mr);
-    if (status != UCS_OK) {
-        goto err_free_memh;
+#if HAVE_IBV_EXP_DM
+    if (flags & UCT_MD_MEM_FLAG_DM) {
```

Comment:
pls exports this code to helper function

Reply:
done